library/std/src/panicking.rs

```diff
@@ -255,18 +258,116 @@ fn default_hook(info: &PanicInfo<'_>) {
     let name = thread.as_ref().and_then(|t| t.name()).unwrap_or("<unnamed>");
 
     let write = |err: &mut dyn crate::io::Write| {
-        let _ = writeln!(err, "thread '{name}' panicked at '{msg}', {location}");
-
         static FIRST_PANIC: AtomicBool = AtomicBool::new(true);
 
+        let mut print_snippet = || -> Result<(), crate::io::Error> {
+            let filename = location.file();
+            let path = PathBuf::from(filename);
+            // We check if `path` is a symlink to avoid people creating a symlink for a file they
+            // don't have permissions, but that the running binary does, to use it as an
+            // exfiltration vector.
+            let mut path_is_symlink = false;
+            let mut path_check = path.clone();
+            loop {
+                if path_check.is_symlink() {
```

Comment:
This check can be circumvented with some good timing.

And it won't work if any of the parent directories is a symlink but not the final component. E.g. `/src/symlink/main.rs`.